vdb/__init__.py

```diff
@@ -1711,29 +1712,76 @@ def do_bpfile(self, line):
 
     def do_bpedit(self, line):
         """
-        Manipulcate the python code that will be run for a given
-        breakpoint by ID.  (Also the way to view the code).
+        Modify a given breakpoint.
+        * Manipulate the python code that will be run for a given
+          breakpoint by ID.
+        * Also the way to view the code
+        * Modify characteristics of the breakpoint (eg. FastBreak)
 
-        Usage: bpedit <id> ["optionally new code"]
+        Usage: bpedit [options] <id> ["optionally new code"]
+
+        where options include:
+           -F  toggles a breakpoint's FastBreak mode
+           -S  toggles a breakpoint's Silent mode (prints a consolem message on break)
+           -V  prints more metadata about a given breakpoint (default is just the code)
 
         NOTE: Your code must be surrounded by "s and may not
         contain any "s
         """
         argv = e_cli.splitargs(line)
+        verbose = False
+        try:
+            opts, args = getopt(argv, 'FSV')
+        except:
+            return self.do_help('bpedit')
+
         if len(argv) == 0:
             return self.do_help("bpedit")
-        bpid = int(argv[0])
 
-        if len(argv) == 2:
-            self.trace.setBreakpointCode(bpid, argv[1])
+        bpid = int(args[0])
+
+        bp = self.trace.getBreakpoint(bpid)
+        if bp is None:
+            self.vprint("Breakpoint %d does not exist!" % bpid)
+            return
+
+        for opt, optarg in opts:
+            if opt == '-V':
+                # print all the metadata
+                verbose = True
 
+            elif opt == '-S':
+                bp.silent = not bp.silent
+                bp = self.trace.getBreakpoint(bpid)
```

Comment:
why do we re-retrieve bp here and below?

Reply:
this was to ensure that everything worked (modifying the in-place breakpoint).  i can remove this code now.  good catch.